app/src/main/kotlin/com/wire/android/ui/authentication/create/common/CreateAccountBaseViewModel.kt

```diff
@@ -196,21 +201,52 @@ abstract class CreateAccountBaseViewModel(
                 CreateAccountFlowType.CreateTeam -> TODO()
             }
 
-            val codeError = registerAccountUseCase(registerParam, serverConfig)
-                .let {
-                    if (it is RegisterResult.Success) // TODO what if user creates an account but doesn't register a new device?
-                        clientScopeProviderFactory.create(it.value.second.userId).clientScope.register(
-                            password = registerParam.password,
-                            capabilities = null
-                        ).toCodeError()
-                    else it.toCodeError()
+            val (userInfo, session) = registerAccountUseCase(registerParam, serverConfig).let {
+                when (it) {
+                    is RegisterResult.Failure -> {
+                        updateCodeErrorState(it.toCodeError())
+                        return@launch
+                    }
+                    is RegisterResult.Success -> it.value
                 }
-            val isSuccess = codeError is CreateAccountCodeViewState.CodeError.None
-            codeState = codeState.copy(loading = false, error = codeError)
-            if (isSuccess) onCodeSuccess()
+            }
+            val storedUserId = addAuthenticatedUser(session, false).let {
+                when (it) {
+                    is AddAuthenticatedUserUseCase.Result.Failure -> {
+                        updateCodeErrorState(it.toCodeError())
+                        return@launch
+                    }
+                    is AddAuthenticatedUserUseCase.Result.Success -> it.userId
+                }
+            }
+            registerClient(storedUserId, registerParam.password).let {
+                when (it) {
+                    is RegisterClientResult.Failure -> {
+                        updateCodeErrorState(it.toCodeError())
+                        return@launch
+                    }
+                    is RegisterClientResult.Success -> onCodeSuccess()
+                }
+            }
         }
     }
 
+    private fun updateCodeErrorState(codeError: CreateAccountCodeViewState.CodeError) {
+        codeState = if (codeError is CreateAccountCodeViewState.CodeError.None) {
+            codeState.copy(error = codeError)
```

Comment:
Why not clear the `loading` when the execution was successful and there is error `None`? Maybe it should have a separate parameter which decides whether `loading` should be true or false? 
Right now I think it's even not possible to get None here, as only Failures are being converted to CodeError, right?

Reply:
Yep, only Failures are mapped to LoginError, is that an issue?

Reply:
It shouldn't be an issue, I just wanted to confirm that, we don't allow the user to go back to the CodeScreen so it shouldn't cause any troubles, but if we include `None` here even if we don't allow it to be passed then maybe we could even simplify it to be just:
```
private fun updateCodeErrorState(codeError: CreateAccountCodeViewState.CodeError) {
    codeState = codeState.copy(loading = false, error = codeError)
}
```
but then maybe we don't even need a separate method for that since it becomes a one-liner 😄